tmpl/blog/introducing-charrua-dhcp.md

```diff
@@ -0,0 +1,219 @@
+Almost every network needs to support DHCP, that is, a way for clients to
+request network parameters from the environment, common parameters are an IP
+address, a network mask, a default gateway and so on.
+
+DHCP can be seen as a critical security component, since it deals usually with
+unauthenticated/unknown peers, therefore it is of special interest to run a
+server as a self-contained Mirage VM.
+
+[Charrua](http://www.github.com/haesbaert/charrua-core) is a DHCP implementation
+written in OCaml, it started off as an excuse to learn more about the language,
+while in development it got picked up on the Mirage mailing lists and became one
+of the [Pioneer
+Projects](https://github.com/mirage/mirage-www/wiki/Pioneer-Projects).
+
+The name `Charrua` is a reference to the, now extinct, semi-nomadic people of
+southern South America, nowadays it is also used to refer to Uruguayan
+nationals. The logic is that DHCP handles dynamic (hence nomadic) clients.
+
+The library is platform agnostic and works outside of Mirage as well, it
+provides two main modules:
+[Dhcp_wire](http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html) and
+[Dhcp_server](http://haesbaert.github.io/charrua-core/api/Dhcp_server.html).
+
+### Dhcp_wire
+
+[Dhcp_wire](http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html) provides
+basic functions for dealing with the protocol, essentialy
+marshalling/unmarshalling and helpers for dealing with the various DHCP options.
+
+The central record type of
+[Dhcp_wire](http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html) is a
+[pkt](http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html#TYPEpkt), which
+represents a full DHCP packet, including layer 2 and layer 3 data as well as the
+many possible DHCP options. The most important functions are:
+
+```ocaml
+val pkt_of_buf : Cstruct.t -> int -> [> `Error of string | `Ok of pkt ]
+val buf_of_pkt : pkt -> Cstruct.t
+```
+
+[pkt_of_buf]
+(http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html#VALpkt_of_buf) takes
+a [Cstruct.t] (https://github.com/mirage/ocaml-cstruct) buffer and a length, it
+then attempts to build a DHCP packet, unknown DHCP options are ignored, invalid
+options or malformed data is not accepted and you get a `Error of string.
+
```

Comment:
What you might want here is `` `Error of string`` (two backticks, space, backtick, Error.., two backticks [since the backtick Error is the actual OCaml code])